<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Entorno de trabajo</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="Final_rBLAST_code_files/libs/clipboard/clipboard.min.js"></script>
<script src="Final_rBLAST_code_files/libs/quarto-html/quarto.js"></script>
<script src="Final_rBLAST_code_files/libs/quarto-html/popper.min.js"></script>
<script src="Final_rBLAST_code_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Final_rBLAST_code_files/libs/quarto-html/anchor.min.js"></script>
<link href="Final_rBLAST_code_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Final_rBLAST_code_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Final_rBLAST_code_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Final_rBLAST_code_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Final_rBLAST_code_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Entorno de trabajo</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="instrucciones-de-uso" class="level2">
<h2 class="anchored" data-anchor-id="instrucciones-de-uso">Instrucciones de uso</h2>
<section id="pasos-de-instalación" class="level3">
<h3 class="anchored" data-anchor-id="pasos-de-instalación">Pasos de instalación</h3>
<p>Es muy importante crear una carpeta que almacene todos los archivos y dependencias necesarias para ejecutar el <strong>script</strong> de la mejor manera. Se recomienda definir el directorio de trabajo y seguir la siguiente estructura de carpetas:</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Carpetas_ficheros.jpg" class="img-fluid figure-img"></p>
</figure>
</div>
<p>Posterior a la creación de carpetas es necesario descomprimir los archivos para su lectura. Es muy importante mencionar que esta librería solamente acepta archivos fasta. <strong>Se recomienda crear una carpeta cada vez que se requiera descomprimir un archivo tar.gz o zip.</strong></p>
<ol type="1">
<li><p>Debe instalarse el algoritmo del blast de la página del NCBI. Puede descargarse en el siguiente enlance: <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download" class="uri">https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download</a></p></li>
<li><p>Posterior de haber instalado el algoritmo, es posible instalar la librería rBLAST. Debe prestar atención a las instrucciones que se desplegarán en la consola.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Cargar el paquete devtools</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(devtools)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: usethis</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Instalar el paquete rblast mediante la librería devtools</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"mhahsler/rBLAST"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Skipping install of 'rBLAST' from a github remote, the SHA1 (c2007a17) has not changed since last install.
  Use `force = TRUE` to force installation</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span> (rBLAST)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: Biostrings</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: BiocGenerics</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'BiocGenerics'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    IQR, mad, sd, var, xtabs</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    anyDuplicated, aperm, append, as.data.frame, basename, cbind,
    colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,
    get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,
    match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,
    Position, rank, rbind, Reduce, rownames, sapply, setdiff, sort,
    table, tapply, union, unique, unsplit, which.max, which.min</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: S4Vectors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: stats4</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'S4Vectors'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    expand.grid, I, unname</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: IRanges</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'IRanges'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:grDevices':

    windows</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: XVector</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: GenomeInfoDb</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'Biostrings'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:base':

    strsplit</code></pre>
</div>
</div></li>
</ol>
<p>En caso de no funcionar es posible realizar la instalación a través de otra repositorio usando el siguiente código:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">'rBLAST'</span>, <span class="at">repos =</span> <span class="st">'https://mhahsler.r-universe.dev'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'rBLAST' is in use and will not be installed</code></pre>
</div>
</div>
<ol start="3" type="1">
<li>Debe descargar la base de datos de interés. En este entorno de trabajo puede usarse cualquier base de datos, siempre y cuando se descargue correctamente. Cualquier base de datos descargada es el <strong><em>input</em></strong> necesario para poder realizar las consultas de la secuencias. En este ejemplo se usará la base de datos de bacterias (16S), descargada de: <a href="https://ftp.ncbi.nlm.nih.gov/blast/db/16S_ribosomal_RNA.tar.gz" class="uri">https://ftp.ncbi.nlm.nih.gov/blast/db/16S_ribosomal_RNA.tar.gz</a> o puede ejecutar la siguiente línea de comando:</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">download.file</span>(<span class="st">"https://ftp.ncbi.nlm.nih.gov/blast/db/16S_ribosomal_RNA.tar.gz"</span>,</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>   <span class="st">"16S_ribosomal_RNA.tar.gz"</span>, <span class="at">mode=</span><span class="st">'wb'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Es posible descargar otras bases de datos siguiendo este enlance: <a href="https://ftp.ncbi.nlm.nih.gov/blast/db/">Bases de datos NCBI</a>, recuerde realizar la creación de todas la carpetas/ficheros, el procedimiento es el mismo que al anterior.</p>
<p><strong>Recuerde definir los ficheros desde el incio de la ejeción del código, usando el comando <code>setwd()</code>.</strong></p>
<p>Para descomprimirlo mendiante el lenguaje de programación R debe ejecutarse el siguiente código:(también se puede hacer el mismo procedimiento utilizando la interfaz gráfica o en su defecto windows):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">untar</span>(<span class="st">"16S_ribosomal_RNA.tar.gz"</span>, <span class="at">exdir=</span><span class="st">"16SMicrobialDB"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="instrucciones-de-ejecución-para-el-framework" class="level1">
<h1>Instrucciones de ejecución para el framework</h1>
<section id="pasos-de-ejecución" class="level3">
<h3 class="anchored" data-anchor-id="pasos-de-ejecución">Pasos de ejecución</h3>
<ol start="4" type="1">
<li>Cargar la librería <code>rBLAST.</code></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rBLAST)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ol start="5" type="1">
<li>Cargar la secuencia de interés.</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Esta secuencia es la que se va a consultar</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>seq <span class="ot">&lt;-</span> <span class="fu">readRNAStringSet</span>(<span class="fu">system.file</span>(<span class="st">"examples/RNA_example.fasta"</span>,</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>                        <span class="at">package=</span><span class="st">"rBLAST"</span>))</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>seq</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>RNAStringSet object of length 5:
    width seq                                               names               
[1]  1481 AGAGUUUGAUCCUGGCUCAGAAC...GGUGAAGUCGUAACAAGGUAACC 1675 AB015560.1 d...
[2]  1404 GCUGGCGGCAGGCCUAACACAUG...CACGGUAAGGUCAGCGACUGGGG 4399 D14432.1 Rho...
[3]  1426 GGAAUGCUNAACACAUGCAAGUC...AACAAGGUAGCCGUAGGGGAACC 4403 X72908.1 Ros...
[4]  1362 GCUGGCGGAAUGCUUAACACAUG...UACCUUAGGUGUCUAGGCUAACC 4404 AF173825.1 A...
[5]  1458 AGAGUUUGAUUAUGGCUCAGAGC...UGAAGUCGUAACAAGGUAACCGU 4411 Y07647.2 Dre...</code></pre>
</div>
</div>
<ol start="6" type="1">
<li>Cargar la base de datos, la cual se va a consultar. En este caso es 16S_ribosomal, usando la función <code>blast</code></li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>bl <span class="ot">&lt;-</span> <span class="fu">blast</span>(<span class="at">db=</span><span class="st">"./16SMicrobialDB/16S_ribosomal_RNA"</span>)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>bl</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>BLAST Database
Location: C:\Users\anton\Documents\rBlast_AF0219\16SMicrobialDB/16S_ribosomal_RNA 
BLAST Type: blastn 
Database: 16S ribosomal RNA (Bacteria and Archaea type strains)
    23,104 sequences; 33,530,608 total bases

Date: Oct 1, 2022  5:36 AM  Longest sequence: 3,600 bases

BLASTDB Version: 5

Volumes:
    C:\Users\anton\Documents\rBlast_AF0219\16SMicrobialDB\16S_ribosomal_RNA</code></pre>
</div>
</div>
<ol start="7" type="1">
<li>Realizar la consulta de la secuencia en la base de datos</li>
</ol>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Según la base de datos consultada, la secuencia tiene similitud con los siguientes registros.</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>cl <span class="ot">&lt;-</span> <span class="fu">predict</span>(bl, seq[<span class="dv">1</span>,])</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>cl[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  QueryID   SubjectID Perc.Ident Alignment.Length Mismatches Gap.Openings
1    1675 NR_104821.1     90.816             1459        124            8
2    1675 NR_117601.1     85.896             1255        160           15
3    1675 NR_117601.1     93.846               65          4            0
4    1675 NR_074549.1     85.896             1255        160           15
5    1675 NR_074549.1     94.203               69          4            0
  Q.start Q.end S.start S.end        E Bits
1      16  1468       5  1459 0.00e+00 1943
2     229  1478     237  1479 0.00e+00 1321
3       5    69       1    65 7.61e-20   99
4     229  1478     241  1483 0.00e+00 1321
5       1    69       1    69 4.55e-22  106</code></pre>
</div>
</div>
</section>
</section>
<section id="ejemplo-de-uso" class="level1">
<h1>Ejemplo de uso</h1>
<p>Se usó la siguiente secuencia a consultar nombrada como sequence2 <a href="https://www.ncbi.nlm.nih.gov/nuccore/LN624761.1" class="uri">https://www.ncbi.nlm.nih.gov/nuccore/LN624761.1</a></p>
</section>
<section id="artículo-científico" class="level1">
<h1>Artículo científico</h1>
<section id="erwinia-endophytica-sp.-nov.-isolated-from-potato-solanum-tuberosum-l.-stems" class="level2">
<h2 class="anchored" data-anchor-id="erwinia-endophytica-sp.-nov.-isolated-from-potato-solanum-tuberosum-l.-stems">Erwinia endophytica sp. nov., isolated from potato (Solanum tuberosum L.) stems</h2>
<p><a href="https://www.microbiologyresearch.org/content/journal/ijsem/10.1099/ijsem.0.000820" class="uri">https://www.microbiologyresearch.org/content/journal/ijsem/10.1099/ijsem.0.000820</a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rBLAST)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>seq1 <span class="ot">&lt;-</span> <span class="fu">readDNAStringSet</span>(<span class="st">"C:/Users/anton/Documents/rBlast_AF0219/Secuencias_fasta/19_nov_2022/sequence.fasta"</span>)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>seq1</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DNAStringSet object of length 1:
    width seq                                               names               
[1]  1454 CTGGCTCAGATTGAACGCTGGCG...GGGAGGGCGCTTACCACTTTGGA LN624761.1 Erwini...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>cl <span class="ot">&lt;-</span> <span class="fu">predict</span>(bl, seq1[<span class="dv">1</span>,])</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a><span class="co">#los primeros 5 resultados</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>cl <span class="ot">&lt;-</span> cl[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,]</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(cl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<table class="table table-sm table-striped">
<colgroup>
<col style="width: 9%">
<col style="width: 10%">
<col style="width: 9%">
<col style="width: 15%">
<col style="width: 9%">
<col style="width: 11%">
<col style="width: 7%">
<col style="width: 5%">
<col style="width: 7%">
<col style="width: 5%">
<col style="width: 2%">
<col style="width: 4%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">QueryID</th>
<th style="text-align: left;">SubjectID</th>
<th style="text-align: right;">Perc.Ident</th>
<th style="text-align: right;">Alignment.Length</th>
<th style="text-align: right;">Mismatches</th>
<th style="text-align: right;">Gap.Openings</th>
<th style="text-align: right;">Q.start</th>
<th style="text-align: right;">Q.end</th>
<th style="text-align: right;">S.start</th>
<th style="text-align: right;">S.end</th>
<th style="text-align: right;">E</th>
<th style="text-align: right;">Bits</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">LN624761.1</td>
<td style="text-align: left;">NR_148650.1</td>
<td style="text-align: right;">100.000</td>
<td style="text-align: right;">1454</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1454</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1454</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2686</td>
</tr>
<tr class="even">
<td style="text-align: left;">LN624761.1</td>
<td style="text-align: left;">NR_104932.1</td>
<td style="text-align: right;">98.624</td>
<td style="text-align: right;">1453</td>
<td style="text-align: right;">18</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1452</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">1465</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2571</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LN624761.1</td>
<td style="text-align: left;">NR_074869.1</td>
<td style="text-align: right;">98.624</td>
<td style="text-align: right;">1453</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">3</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1452</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">1470</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2569</td>
</tr>
<tr class="even">
<td style="text-align: left;">LN624761.1</td>
<td style="text-align: left;">NR_042422.1</td>
<td style="text-align: right;">98.555</td>
<td style="text-align: right;">1453</td>
<td style="text-align: right;">17</td>
<td style="text-align: right;">4</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">1452</td>
<td style="text-align: right;">13</td>
<td style="text-align: right;">1463</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2564</td>
</tr>
<tr class="odd">
<td style="text-align: left;">LN624761.1</td>
<td style="text-align: left;">NR_118431.1</td>
<td style="text-align: right;">98.547</td>
<td style="text-align: right;">1445</td>
<td style="text-align: right;">19</td>
<td style="text-align: right;">2</td>
<td style="text-align: right;">10</td>
<td style="text-align: right;">1452</td>
<td style="text-align: right;">1</td>
<td style="text-align: right;">1445</td>
<td style="text-align: right;">0</td>
<td style="text-align: right;">2553</td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>